apps/web/src/contexts/auth-context.tsx

```diff
@@ -405,26 +418,34 @@ export function useAuthProvider() {
       return
     }
 
-    const auth = getAuth(firebaseApp)
-    const unsubscribe = auth.onAuthStateChanged(async (user) => {
-      dispatch(authActions.setLoading())
-      if (user) {
-        await reloadProfile()
-      } else {
-        dispatch(authActions.setAnonymous())
-      }
-    })
-
-    getRedirectResult(auth).then(() => {
-      const method =
-        auth.currentUser?.providerData[0].providerId === 'password'
-          ? 'email'
-          : 'google'
-      dispatch(authActions.setMethod(method))
-      analytics.login(method)
-    })
+    let unsubscribe: Unsubscribe | null = null
+
+    ;(async () => {
```

Comment:
Out of curiosity, what is this semi doing? I'm presuming it's a convention I'm not aware of.

Reply:
Prevents prettier from mangling this together with the previous line, since semicolon is the "terminator" for JS statements.

Reply:
ahhhh prettier ;(